packages/stex-react-renderer/src/lib/SubProblemAnswer.tsx

```diff
@@ -231,6 +237,20 @@ export function SubProblemAnswer({
           </Box>
         )}
       </Box>
+      {isGrading && isFrozen && subProblem.solution && (
```

Comment:
Don't copy paste. Create a variable before the return statement like so:
```
  const solutionBox = isFrozen && subProblem.solution ? (<Box style={{...
  </Box>) : <></>;
  ```
  and then show this solutionBox on the top or bottom based on `isGrading` variable